### PR TITLE
Add Getting Started docs

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below
not needed
---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This adds some basic guidelines for prompting, and some ideas on where OpenHands is useful

---
**Link of any specific issues this addresses**

### CODE REVIEW

File: docs/modules/usage/getting-started.mdx [hunk 1]
```diff
@@ -1,67 +1,103 @@
...
+> Currently the email field in the `/subscribe` endpoint is rejecting .io domains. Please fix this.
+
+> The `search_widgets` function in ./app.py is doing a case-sensitive search. Please make it case-insensitive.
+
+It often helps to do test-driven development when bugfixing with an agent.
+You can ask the agent to write a new test, and then iterate until it fixes the bug:
+
+> The `hello` function crashes on the empty string. Please write a test that reproduces this bug, then fix the code so it passes.
+
+## More
```
Comment (neubig) [inline docs/modules/usage/getting-started.mdx:97]: Maybe we should link to this page? https://docs.all-hands.dev/modules/usage/prompting-best-practices

Comment (rbren) [reply docs/modules/usage/getting-started.mdx:97]: Oh hey look at that--let me combine

Comment (rbren) [reply docs/modules/usage/getting-started.mdx:97]: Done! Moved the best practice doc up to the top level too

File: docs/modules/usage/installation.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+sidebar_position: 2
+---
```
Comment (mamoodi) [inline docs/modules/usage/installation.mdx:3]: Not needed

**Review by tofarr — APPROVED** [review]

**Pull request merged**